src/registrar/models/domain.py

```diff
@@ -267,25 +373,234 @@ def administrative_contact(self) -> PublicContact:
+        self._make_contact_in_registry(contact=contact)
+        self._update_domain_with_contact(contact, rem=False)
 
-    @Cache
-    def security_contact(self) -> PublicContact:
-        """Get or set the security contact for this domain."""
-        # TODO: replace this with a real implementation
+    def get_default_security_contact(self):
+        logger.info("getting default sec contact")
         contact = PublicContact.get_default_security()
         contact.domain = self
-        contact.email = "mayor@igorville.gov"
         return contact
 
+    def _update_epp_contact(self, contact: PublicContact):
+        """Sends UpdateContact to update the actual contact object,
+        domain object remains unaffected
+        should be used when changing email address
+        or other contact info on an existing domain
+        """
+        updateContact = commands.UpdateContact(
+            id=contact.registry_id,
+            # type: ignore
+            postal_info=self._make_epp_contact_postal_info(contact=contact),
+            email=contact.email,
+            voice=contact.voice,
+            fax=contact.fax,
+        )  # type: ignore
+
+        try:
+            registry.send(updateContact, cleaned=True)
+        except RegistryError as e:
+            logger.error(
+                "Error updating contact, code was %s error was %s" % (e.code, e)
+            )
+            # TODO - ticket 433 human readable error handling here
+
+    def _update_domain_with_contact(self, contact: PublicContact, rem=False):
+        """adds or removes a contact from a domain"""
+        logger.info(
+            "_update_domain_with_contact() received type %s " % contact.contact_type
+        )
+        domainContact = epp.DomainContact(
+            contact=contact.registry_id, type=contact.contact_type
+        )
+
+        updateDomain = commands.UpdateDomain(name=self.name, add=[domainContact])
+        if rem:
+            updateDomain = commands.UpdateDomain(name=self.name, rem=[domainContact])
+
+        try:
+            registry.send(updateDomain, cleaned=True)
+        except RegistryError as e:
+            logger.error(
+                "Error changing contact on a domain. Error code is %s error was %s"
+                % (e.code, e)
+            )
+            action = "add"
+            if rem:
+                action = "remove"
+
+            raise Exception(
+                "Can't %s the contact of type %s" % (action, contact.contact_type)
+            )
+
+    @Cache
+    def security_contact(self) -> PublicContact:
+        """Get or set the security contact for this domain."""
+
+        # get the contacts: call _get_property(contacts=True)
+        # if contacts exist and security contact is in the contact list
+        # return that contact
+        # else call the setter
+        #   send the public default contact
+        try:
+            contacts = self._get_property("contacts")
+            for contact in contacts:
```

Comment:
Should we be accounting for an array of security_contacts to be returned here? I was under the assumption that, while possible, for our purposes we will only deal with one security_contact (and others in parallel) per domain

Reply:
This is an array of all the contacts, it hasn't retrieved the security one yet. Good question though!